### PR TITLE
Add better javadocs to the ArrayBasedTable

### PR DESCRIPTION
**Pull request by suneet-s**

### CODE REVIEW

**Comment by suneet-s** [comment]
@jon-wei

File: src/main/java/io/imply/druid/indextable/loader/ArrayBasedTable.java [hunk 1]
```diff
@@ -32,12 +32,36 @@
...
    * A list of rows. Each row is a Pair of timestamp and a list of encoded values.
+   *
+   * For the exmaple table in {@link #dimensionDictionary}
+   * col1 col2 col3
+   * A    cat  12
+   * B    cat  23
+   * C    dog  23
+   *
+   * {@code #encodedRows} will look like
+   * [(ts, [0, 0, 0]), (ts, [1, 0, 1]), (ts, [2, 1, 1])]
```
Comment (suneet-s) [inline src/main/java/io/imply/druid/indextable/loader/ArrayBasedTable.java]: When I first implemented this, I wasn't sure what to do with the timestamp function for a row. I now believe that function is not useful and can contribute to a lot of wasted space. I'm going to remove this in my next PR and have the timestamp function return 0. @gianm @jon-wei thoughts about this?

File: src/main/java/io/imply/druid/indextable/loader/ArrayBasedTable.java [hunk 2]
```diff
@@ -32,12 +32,36 @@
...
+   * For example a table called foo
+   * col1 col2 col3
+   *  A    cat  12
+   *  B    cat  23
+   *  C    dog  23
+   *
+   *  will produce a {@link #dimensionDictionary} that looks like
+   *  col1 -> (0, {("A", 0), ("B", 1), ("C", 2)})
+   *  col2 -> (1, {("cat", 0), ("dog", 1)})
+   *  col3 -> (2, {(12, 0), (23, 1)})
```
Comment (suneet-s) [inline src/main/java/io/imply/druid/indextable/loader/ArrayBasedTable.java]: Looking at some profiling results, we see getting the value for a dimensionIndex in the BiMap is expensive. I propose to write our own implementation of the BiMap, that is backed by a FastUtil.Int2ObjectMap. That should reduce the time spent shown in this flamegraph
image

File: src/main/java/io/imply/druid/indextable/loader/ArrayBasedTable.java (hunk 1 above)
Comment (jon-wei) [reply src/main/java/io/imply/druid/indextable/loader/ArrayBasedTable.java]: It's fine to not have a real timestamp, Druid just expects that there is a timestamp column of some sort, the ingestion specs also support filling in the timestamp column with a default value:

```
"timestampSpec": {
          "column": "!!!_no_such_column_!!!",
          "missingValue": "2020-03-01T00:00:00Z"
        },
```

**Pull request merged**